src/server-middleware/healthcheck.js

```diff
@@ -1,4 +1,49 @@
-export default function healthcheck(req, res) {
-  res.setHeader('Content-Type', 'text/plain')
-  res.end('OK')
+import osu from 'node-os-utils'
+
+/**
+ * A Healthcheck which passes if the amount of avaliable memory
+ * is greater than our desired minimum avaliable memory percentage.
+ *
+ * Useful in the event of memory leaks, in particular.
+ *
+ * @type {import('@nuxt/types').ServerMiddleware}
+ */
+export default function Healthcheck(_, res) {
+  osu.mem
+    .info()
+    .then((info) => {
+      const avaliableMemory = info.freeMemPercentage
+
+      /**
+       * Server Middleware do not have access to any context or configuration,
+       * so we use a raw environment variable and set a default here, instead
+       * of in `nuxt.config.ts` or `utils/env.ts`.
+       */
+      const desiredAvaliableMemory = Number(
```

Comment:
Do we need to add a check so that we accidentally don't set the desiredAvailableMemory to less than 0 or more than 100?

Reply:
I'm not sure, I did add a comment to the env file about this but I don't think we need to validate it in the code.